src/NUnitFramework/tests/Internal/TestSuiteCopyTests.cs

```diff
@@ -0,0 +1,69 @@
+using NUnit.Framework.Api;
+using NUnit.TestData.OneTimeSetUpTearDownData;
+using NUnit.TestUtilities;
+using System.IO;
+
+namespace NUnit.Framework.Internal
+{
+    // Tests that Copy is properly overridden for all types extending TestSuite - #3171
+    public class TestSuiteCopyTests
+    {
+        [Test]
+        public void CopyTestSuiteReturnsCorrectType()
+        {
+            TestSuite testSuite =
+                new TestSuite(new TypeWrapper(typeof(NUnit.TestData.ParameterizedTestFixture)));
+            var copiedTestSuite = testSuite.Copy(TestFilter.Empty);
+            Assert.AreEqual(copiedTestSuite.GetType(), typeof(TestSuite));
```

Comment:
Expected and actual are backwards in each test.

This is exactly why I dislike the Assert.AreEqual syntax. Elsewhere we use `Assert.That` and I personally find that more intuitive: `Assert.That(copy, Is.TypeOf<TestSuite>());`

Reply:
@cjconrad2 Sorry to bug you since I'm sure you're busy, but I want to make sure that my requested change was clear. Either swapping the AreEqual arguments in each test or switching to Assert.That syntax is fine.

Reply:
Sorry about the delay all, and thank you for your patience! I had a long vacation and I have spent the last week or so catching up. I updated the tests and decided that I also prefer the Assert.That syntax more. Thanks for suggesting that! Hopefully we can merge soon

Reply:
The delay is no problem for us. I wanted to make sure I wasn't the cause of the delay, but I'm sorry I interrupted your vacation!